src/lib/builders/table-of-contents/create.ts

```diff
@@ -0,0 +1,259 @@
+                // Only add the observed element to the visibleElementIdxs list if it isn't added yet.
+
+                if (tempVisibleElementIdxs.indexOf(el_idx) === -1) {
+                    tempVisibleElementIdxs = [...tempVisibleElementIdxs, el_idx];
+                    visibleElementIdxs.set(tempVisibleElementIdxs);
+
+                    // Only add active parents if parent headings should be highlighted.
+                    activeParentIdxs.set((tocType === 'highest-parents' || tocType === 'lowest-parents') && headingParentsLU[toc_idx]
+                            ? [...get(activeParentIdxs), ...(<number[]>headingParentsLU[toc_idx])]
+                            : []);
+                }
+            } else {
+                // Remove the observed element from the visibleElementIdxs list if the intersection ratio is below the threshold.
+                tempVisibleElementIdxs = tempVisibleElementIdxs.filter((item) => item !== el_idx);
+                visibleElementIdxs.set(tempVisibleElementIdxs);
+
+                // Remove all parents of obsIndex from the activeParentIdxs list.
+                if ((tocType === 'highest-parents' || tocType === 'lowest-parents') && headingParentsLU[toc_idx]) {
+                    headingParentsLU[toc_idx]?.forEach((parent: number) => {
+                        const tempActiveParentIdxs = get(activeParentIdxs);
+                        const index = tempActiveParentIdxs.indexOf(parent);
+                        tempActiveParentIdxs.splice(index, 1);
+                        activeParentIdxs.set(tempActiveParentIdxs);
+                    });
+                }
+            }
+        }
+
+        const allActiveHeaderIdxs = Array.from(new Set(get(visibleElementIdxs).map((idx) => elementHeadingLU[idx])));
+
+        let activeHeaderIdxs: number[];
+
+        if (allActiveHeaderIdxs.length === 0) {
+            activeHeaderIdxs = [];
+        } else if (tocType === 'highest') {
+            activeHeaderIdxs = [Math.min(...allActiveHeaderIdxs)];
+        } else if (tocType === 'lowest') {
+            activeHeaderIdxs = [Math.max(...allActiveHeaderIdxs)];
+        } else if (tocType === 'all-active') {
+            activeHeaderIdxs = allActiveHeaderIdxs;
+        } else {
+            const activeHeaderIdx = tocType === 'highest-parents' ? Math.min(...allActiveHeaderIdxs) : Math.max(...allActiveHeaderIdxs);
+
+            if (headingParentsLU[activeHeaderIdx]) {
+                activeHeaderIdxs = [...<[]>headingParentsLU[activeHeaderIdx], activeHeaderIdx];
+            } else {
+                activeHeaderIdxs = [activeHeaderIdx];
+            }
+        }
+
+        // Set store to active indexes.
+        activeHeadingIdxs.set(activeHeaderIdxs);
+    }
+
+    const item = builder('', {
+        returned: () => {
+            return {
+                role: 'link',
+                tabindex: 0
+            }
+        },
+        action: (node: HTMLElement, params: NavItemProps) => {
```

Comment:
I definitely think we should enforce to make this an A tag. Also, I'd change this so the params don't come from the action, but from the store (e.g. `melt={$item(id)}` or `{...$item(id)} use:item`. This way we can use the melt attribute. 

To get this value from within the action, you can store the id in a data attr and get it later

Reply:
@TGlide I'm still trying to get familiar with the API, so with melt attribute you mean that the `id` passed to the store will be added as an attribute to the element? And then inside the action we can grab the id from the attribute right? What is the name of the attribute I should use for this?

Reply:
Exactly! Maybe data-id?